### PR TITLE
ROC-2665: Add gradle and other config files

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,76 @@
-# service-authorisation-access-client
+# service-auth-provider-java-client
+
+This is the backend service for Civil Money Claims.
```
Comment (timja-kainos) [inline README.md]: might want to update

File: README.md [hunk 2]
```diff
@@ -1 +1,76 @@
...
+```bash
+    ./gradlew build
+```
+## Configuration
+
+To use the services provided by this clients, they need to be instantiated in spring Configuration class
+```java
+   @Configuration
+   @Lazy
+   @EnableFeignClients(basePackageClasses = ServiceAuthorisationApi.class)
```
Comment (timja-kainos) [inline README.md]: does this work?

Comment (dharmendrak) [reply README.md]: tested code

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+group 'uk.gov.hmcts.reform'
+version '0.0.1'
+
+checkstyle {
+    maxWarnings = 0
+    toolVersion = '8.3'
+    // need to set configDir to rootDir otherwise submodule will use submodule/config/checkstyle
+//    configDir = new File(rootDir, 'config/checkstyle')
```
Comment (timja-kainos) [inline build.gradle]: hmm?

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+publishing {
+    publications {
+        Main(MavenPublication) {
+            from components.java
+            artifact sourcesJar
+            artifact javadocJar
+            groupId project.group
+            artifactId 'service-auth-provider-java-client'
```
Comment (timja-kainos) [inline build.gradle]: no java here!!!!

File: build.gradle (hunk 3 above)
Comment (dharmendrak) [reply build.gradle]: I'll remove

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            version project.version
+
+            pom.withXml {
+                def root = asNode()
+                root.appendNode('description', 'Client library for communicating with idam service authorisation access')
```
Comment (timja-kainos) [inline build.gradle]: how about: idam service auth provider api?

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+                root.appendNode('name', 'Service authorisation access client')
```
Comment (timja-kainos) [inline build.gradle]: what about: Service auth provider api client

File: build.gradle [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+bintray {
+    user = System.getenv('BINTRAY_USER')
+    key = System.getenv('BINTRAY_KEY')
+    publications = ['Main']
+    publish = true
+    pkg {
+        repo = 'hmcts-maven'
+        name = 'service-auth-provider-java-client'
```
Comment (timja-kainos) [inline build.gradle]: no java!

File: settings.gradle [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+ */
+
+/*
+// To declare projects as part of a multi-project build use the 'include' method
+include 'shared'
+include 'api'
+include 'services:webservice'
+*/
+
+rootProject.name = 'service-auth-provider-java-client'
```
Comment (timja-kainos) [inline settings.gradle]: no java here to I would think?

Comment (dharmendrak) [reply settings.gradle]: done

File: build.gradle (hunk 7 above)
Comment (dharmendrak) [reply build.gradle]: done

File: build.gradle (hunk 6 above)
Comment (dharmendrak) [reply build.gradle]: done

File: build.gradle (hunk 5 above)
Comment (dharmendrak) [reply build.gradle]: done

File: build.gradle (hunk 4 above)
Comment (dharmendrak) [reply build.gradle]: done

File: README.md (hunk 1 above)
Comment (dharmendrak) [reply README.md]: reworded

File: README.md [hunk 9]
```diff
@@ -1 +1,76 @@
-# service-authorisation-access-client
+# service-auth-provider-java-client
+
+This is the Service auth provider client for Civil Money Claims.
```
Comment (timja-kainos) [inline README.md]: its not for cmc its for everyone

Comment (dharmendrak) [reply README.md]: done

File: README.md [hunk 10]
```diff
@@ -1 +1,76 @@
-# service-authorisation-access-client
+# service-auth-provider-java-client
+
+This is the client library for the service-auth-provider api micro service.
+The service provides a methods to generate s2s auth token for a micro service and caches it for 4 hours.
```
Comment (rkondratowicz) [inline README.md]: typo. either 'a method' or 'methods'

**Pull request merged**